docs/reference/java/generated-code.md

```diff
@@ -0,0 +1,319 @@
+---
+bodyclass: docs
+headline: Java Generated Code Reference
+layout: docs
+type: markdown
+---
+
+# Java Generated Code Reference
+
+## Packages
+
+For each service defined in a .proto file, the Java code generation produces a
+Java class. The class name is the service's name suffixed by `Grpc`. The package
+for the generated code is specified in the .proto file using the `java_package`
+option.
+
+For example, if `ServiceName` is defined in a .proto file containing the
+following:
+
+```
```

Comment:
please add `java` after the backtick to this and other multiline code examples in the file to ensure proper highlighting